File: projet-crud/src/app/models/UtilisateurDTO.ts
import { TerrainDTO } from './TerrainDTO';

export interface UtilisateurDTO {
  id?: number;
  nom: string;
  prenom?: string;
  mail: string;
  password: string;
  username: string;
  terrains?: TerrainDTO[];
  role?: string;
}

File: poly-gemimi/src/main/java/fr/polytech/poly_gemimi/dto/UtilisateurDTO.java
package fr.polytech.poly_gemimi.dto;

import fr.polytech.poly_gemimi.entity.Terrain;
import fr.polytech.poly_gemimi.entity.Utilisateur;
import lombok.*;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UtilisateurDTO {
    private Long id;
    private String nom;
    private String prenom;
    private String mail;
    private String password;
    private String username;
    private Set<TerrainDTO> terrains;
    private String role;
}
